server/plugin/api.go

```diff
@@ -1258,23 +1471,295 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+		State:       &(status),
+		StateReason: &statusReason,
+	}
+
+	issue, resp, err := githubClient.Issues.Edit(c.Ctx, owner, repo, numberInt, githubIssue)
+	if err != nil {
+		if resp != nil && resp.Response.StatusCode == http.StatusGone {
+			p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "Issues are disabled on this repository.", StatusCode: http.StatusMethodNotAllowed})
+			return
+		}
+
+		c.Log.WithError(err).Warnf("Failed to update the issue")
+		p.writeAPIError(w, &APIErrorResponse{
+			ID: "",
+			Message: "failed to update issue: " + getFailReason(resp.StatusCode,
+				repo,
+				currentUsername,
+			),
+			StatusCode: resp.StatusCode,
+		})
+		return
+	}
+	var permalinkReplyMessage string
+	switch statusReason {
+	case "completed":
+		permalinkReplyMessage = fmt.Sprintf("Issue closed as not completed [#%v](%v)", numberInt, issue.GetHTMLURL())
+	case "not_planned":
+		permalinkReplyMessage = fmt.Sprintf("Issue closed as not planned [#%v](%v)", numberInt, issue.GetHTMLURL())
+	default:
+		permalinkReplyMessage = fmt.Sprintf("Issue reopend [#%v](%v)", numberInt, issue.GetHTMLURL())
+	}
+
+	rootID := post.Id
+	if post.RootId != "" {
+		// the original post was a reply
+		rootID = post.RootId
+	}
+	reply := &model.Post{
+		Message:   permalinkReplyMessage,
+		ChannelId: post.ChannelId,
+		RootId:    rootID,
+		UserId:    c.UserID,
+	}
+
+	_, appErr := p.API.CreatePost(reply)
+	if appErr != nil {
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to create notification post " + post.Id, StatusCode: http.StatusInternalServerError})
+		return
+	}
+	if status == actionClosed {
+		post.Props[IssueStatus] = Reopen
+	} else {
+		post.Props[IssueStatus] = Close
+	}
+	if _, appErr = p.API.UpdatePost(post); appErr != nil {
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to update post " + post.Id, StatusCode: http.StatusInternalServerError})
+	}
+	p.writeJSON(w, issue)
+}
+
+func (p *Plugin) closeOrReopenIssue(c *UserContext, w http.ResponseWriter, r *http.Request) {
+	req := &CommentAndCloseRequest{}
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		c.Log.WithError(err).Warnf("Error decoding the JSON body")
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a valid JSON object.", StatusCode: http.StatusBadRequest})
+		return
+	}
+	numberInt, err := strconv.Atoi(req.Number)
+	if err != nil {
+		p.writeAPIError(w, &APIErrorResponse{Message: "Invalid param 'number'.", StatusCode: http.StatusBadRequest})
+		return
+	}
+
+	post, appErr := p.API.GetPost(req.PostID)
+	if appErr != nil {
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: fmt.Sprintf("failed to load the post %s", req.PostID), StatusCode: http.StatusInternalServerError})
+		return
+	}
+	if post == nil {
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: fmt.Sprintf("failed to load the post %s : not found", req.PostID), StatusCode: http.StatusNotFound})
+		return
+	}
+
+	_, err = p.getUsername(post.UserId)
```

Comment:
Check this everywhere
```suggestion
	if _, err = p.getUsername(post.UserId); err !=nil {
```